core/src/main/scala/org/apache/spark/storage/BlockManagerMaster.scala

```diff
@@ -230,6 +238,7 @@ class BlockManagerMaster(
     if (driverEndpoint != null && isDriver) {
       tell(StopBlockManagerMaster)
       driverEndpoint = null
+      driverHeartbeatEndPoint = null
```

Comment:
you'd better stop `driverHeartbeatEndPoint` before null it out.

Reply:
fixed in `tell`